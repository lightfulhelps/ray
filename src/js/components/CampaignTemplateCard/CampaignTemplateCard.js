// @flow
import * as React from 'react';
import classNames from 'classnames';
import { Button } from '../..';

type Props = {
  className?: string,
  ctaText: string,
  ctaTextSelected?: string,
  description: string,
  image: string,
  isSelected?: boolean,
  onClick: (SyntheticMouseEvent<>) => void,
  title: string,
};

const CampaignTemplateCard = ({
  className,
  ctaText = 'Use this template',
  ctaTextSelected = 'Selected',
  onClick,
  title,
  description,
  image,
  isSelected,
  ...other
}: Props) => {
  const classes = classNames(
    className,
    'card h-100 rounded-lg',
    { 'campaign-template-card--selected': isSelected },
    'campaign-template-card'
  );

  const imgStyles = {
    backgroundImage: `url(${image})`,
    height: 0,
    paddingBottom: '50%',
    backgroundPosition: 'center',
  };

  const btnProps: any = {
    'data-test-id': 'campaign-template-card-button',
    className: 'mt-2',
    onClick,
    isBlock: true,
    isDisabled: false,
    theme: 'primary',
    children: ctaText,
  };

  if (isSelected) {
    btnProps.isDisabled = true;
    btnProps.theme = 'success';
    btnProps.icon = 'tick';
    btnProps.children = ctaTextSelected;
  }

  return (
    <div {...other} className={classes}>
      <div
        style={imgStyles}
        className="bg-cover bg-gray-900 w-100 rounded-top-lg"
        data-test-id="campaign-template-card-image"
      />
      <div className="p-2 d-flex h-100 flex-column justify-content-between">
        <div>
          <h5 data-test-id="campaign-template-card-title">{title}</h5>
          <div className="text-sm" data-test-id="campaign-template-card-description">
            {description}
          </div>
        </div>
        <Button {...btnProps} />
      </div>
    </div>
  );
};

export default CampaignTemplateCard;
