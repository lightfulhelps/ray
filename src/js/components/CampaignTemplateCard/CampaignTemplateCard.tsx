import * as React from 'react';
import classNames from 'classnames';
import { Button } from '../..';
import { Props as ButtonProps } from '../Button/Button';

type Props = {
  className?: string;
  ctaText?: string;
  ctaTextSelected?: string;
  description: string;
  image: string;
  isSelected?: boolean;
  onClick?: (event: React.MouseEvent) => void;
  title: string;
  [key: string]: any;
};

const CampaignTemplateCard: React.FC<Props> = ({
  className,
  ctaText = 'Use this template',
  ctaTextSelected = 'Selected',
  onClick,
  title,
  description,
  image,
  isSelected,
  ...other
}: Props) => {
  const classes = classNames(
    className,
    'card h-100 rounded-lg',
    { 'campaign-template-card--selected': isSelected },
    'campaign-template-card'
  );

  const imgStyles = {
    backgroundImage: `url(${image})`,
    height: 0,
    paddingBottom: '50%',
    backgroundPosition: 'center',
  };

  const btnProps: ButtonProps = {
    className: 'mt-3',
    onClick,
    isDisabled: false,
    theme: 'primary',
    children: ctaText,
    size: 'lg',
  };

  if (isSelected) {
    btnProps.isDisabled = true;
    btnProps.icon = 'approvalTick';
    btnProps.children = ctaTextSelected;
  }

  return (
    <div {...other} className={classes}>
      <div
        style={imgStyles}
        className="bg-cover bg-gray-900 w-100 campaign-template-card-image"
        data-test-id="campaign-template-card-image"
      />
      <div className="p-3 d-flex h-100 flex-column justify-content-between">
        <div>
          <h4 data-test-id="campaign-template-card-title">{title}</h4>
          <div className="text-sm" data-test-id="campaign-template-card-description">
            {description}
          </div>
        </div>
        <div>
          <Button data-test-id="campaign-template-card-button" {...btnProps} />
        </div>
      </div>
    </div>
  );
};

export default CampaignTemplateCard;
