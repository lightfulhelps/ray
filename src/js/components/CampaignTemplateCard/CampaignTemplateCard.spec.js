import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import CampaignTemplateCard from './CampaignTemplateCard';

const setup = (overrides = {}) => {
  const props = merge(
    {
      title: 'Create your own',
      description: "Don't use a template, do your own thing.",
      image: 'http://foo.com/bar.jpg',
      onClick: () => {},
    },
    overrides
  );
  const wrapper = shallow(<CampaignTemplateCard {...props} />);

  return { wrapper, props };
};

describe('<CampaignTemplateCard />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render when isSelected', () => {
    const { wrapper } = setup({ isSelected: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle className', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('campaign-template-card')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle title', () => {
    const title = 'My title';
    const { wrapper } = setup({ title });

    expect(wrapper.find('[data-test-id="campaign-template-card-title"]').text()).toEqual(title);
  });

  it('should handle description', () => {
    const description = 'My description';
    const { wrapper } = setup({ description });

    expect(wrapper.find('[data-test-id="campaign-template-card-description"]').text()).toEqual(
      description
    );
  });

  it('should handle background image', () => {
    const image = 'bar.com/foo.jpg';
    const { wrapper } = setup({ image });

    const style = wrapper.find('[data-test-id="campaign-template-card-image"]').prop('style');
    expect(style).toHaveProperty('backgroundImage', 'url(bar.com/foo.jpg)');
  });

  it('should run onClick event when button is clicked', () => {
    const onClick = jest.fn();
    const { wrapper } = setup({ onClick });

    wrapper.find('[data-test-id="campaign-template-card-button"]').simulate('click');
    expect(onClick).toBeCalledTimes(1);
  });

  it('should pass through other props', () => {
    const { wrapper } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });
});
