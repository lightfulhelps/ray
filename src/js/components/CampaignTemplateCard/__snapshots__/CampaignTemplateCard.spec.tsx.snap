// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CampaignTemplateCard /> should render 1`] = `
<div
  className="card h-100 rounded-lg campaign-template-card"
>
  <div
    className="bg-cover bg-gray-900 w-100 campaign-template-card-image"
    data-test-id="campaign-template-card-image"
    style={
      Object {
        "backgroundImage": "url(http://foo.com/bar.jpg)",
        "backgroundPosition": "center",
        "height": 0,
        "paddingBottom": "50%",
      }
    }
  />
  <div
    className="p-3 d-flex h-100 flex-column justify-content-between"
  >
    <div>
      <h4
        data-test-id="campaign-template-card-title"
      >
        Create your own
      </h4>
      <div
        className="text-sm"
        data-test-id="campaign-template-card-description"
      >
        Don't use a template, do your own thing.
      </div>
    </div>
    <div>
      <Button
        className="mt-3"
        data-test-id="campaign-template-card-button"
        isDisabled={false}
        onClick={[Function]}
        size="lg"
        theme="primary"
      >
        Use this template
      </Button>
    </div>
  </div>
</div>
`;

exports[`<CampaignTemplateCard /> should render when isSelected 1`] = `
<div
  className="card h-100 rounded-lg campaign-template-card--selected campaign-template-card"
>
  <div
    className="bg-cover bg-gray-900 w-100 campaign-template-card-image"
    data-test-id="campaign-template-card-image"
    style={
      Object {
        "backgroundImage": "url(http://foo.com/bar.jpg)",
        "backgroundPosition": "center",
        "height": 0,
        "paddingBottom": "50%",
      }
    }
  />
  <div
    className="p-3 d-flex h-100 flex-column justify-content-between"
  >
    <div>
      <h4
        data-test-id="campaign-template-card-title"
      >
        Create your own
      </h4>
      <div
        className="text-sm"
        data-test-id="campaign-template-card-description"
      >
        Don't use a template, do your own thing.
      </div>
    </div>
    <div>
      <Button
        className="mt-3"
        data-test-id="campaign-template-card-button"
        icon="approvalTick"
        isDisabled={true}
        onClick={[Function]}
        size="lg"
        theme="primary"
      >
        Selected
      </Button>
    </div>
  </div>
</div>
`;
